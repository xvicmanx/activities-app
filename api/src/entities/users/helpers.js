// @flow

export const getSafeUser = (user: Object) => Object.keys((user.toJSON ? user.toJSON() : user) || {})
  .filter((key) => key !== 'password')
  .reduce((acc, key) => ({
    ...acc,
    [key]: user[key],
  }), {});

export default {
  getSafeUser,
};
