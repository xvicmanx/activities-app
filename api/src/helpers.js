/* eslint-disable max-classes-per-file */
// @flow

/**
 * @module helpers
 */

import type {
  $Request,
  $Response,
} from 'express';
import _ from 'lodash';

import UsersService from './entities/users/service';

const extractTokenFromRequest = (req: $Request) => _.get(
  req,
  'headers.authorization',
  '',
).replace('Bearer ', '');

/**
 * Gets the logged in user using the JWT token
 * @name getLoggedInUser
 * @param {$Response} res - Express response
 * @return {User} logged in user
 */
export const getLoggedInUser = (req: $Request) => {
  const service = new UsersService();
  const token = extractTokenFromRequest(req);

  if (!token) {
    return null;
  }

  return service.findByToken(token);
};

/**
 * Wrapper to make a request auth dependant.
 * If the user is not signed in it will throw 401 error
 * @name authRequired
 * @param {Function} cb - function to wrap
 */
export const authRequired = (cb: Function) => (req: $Request, res: $Response, next: Function) => {
  if (!res.locals.user) {
    res.status(401).json({
      message: 'It is not authorized',
      code: 'NOT_AUTHORIZED',
    });
  } else {
    cb(req, res, next);
  }
};

/**
 * Throw a invalid param error with a 400 status
 * @name invalidParamError
 * @param {$Request} req - Express request
 * @param {$Response} res - Express response
 * @param {string} message - message of the error
 */
export const invalidParamError = (req: $Request, res: $Response, message: string) => {
  res.status(400).json({
    message: `Invalid parameters: ${message}`,
    code: 'INVALID_PARAMETERS',
  });
};

/**
 * Transforms a target to object in case of not being
 * @name asObject
 * @param {mixed} target - target item
 * @return {Object} result
 */
export const asObject = (target: mixed): Object => ((target !== null && typeof target === 'object')
  ? target : {});

/**
 * Errors related to Validation
 * @name ValidationError
 */
export class ValidationError extends Error {
  code: string;

  status: number;

  constructor(code: string, message: string = '') {
    super(message);
    this.name = 'ValidationError';
    this.status = 400;
    this.code = code;
  }
}

/**
 * Errors related to not found objects
 * @name NotFoundError
 */
export class NotFoundError extends Error {
  code: string;

  status: number;

  constructor(message: string = '') {
    super(message);
    this.name = 'NotFoundError';
    this.status = 404;
    this.code = 'NOT_FOUND';
  }
}

/**
 * Throws a validation error
 * @name throwValidationError
 * @throws {ValidationError}
 * @param {string} param - param name
 * @param {string} extraMessage - additional message
 */
export const throwValidationError = (param: string, extraMessage: string = '') => {
  throw new ValidationError(param, extraMessage);
};

/**
 * Throws a validation error
 * @name throwNotFoundError
 * @throws {NotFoundError}
 * @param {string} message - error message
 */
export const throwNotFoundError = (message: string) => {
  throw new NotFoundError(message);
};

/**
 * Handles error returning the correct status
 * @name errorHandler
 * @param {Function} fn - Funtion to handle errors
 */
export const errorHandler = (fn: Function) => async (req: $Request, res: $Response) => {
  try {
    const result = await fn(req, res);
    return result;
  } catch (err) {
    res.status(err.status || 500).json({
      message: err.status ? err.message : 'Unexpected error',
    });
    return null;
  }
};
