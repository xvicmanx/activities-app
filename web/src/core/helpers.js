// @flow

type Direction = 'ascending' | 'descending';

type Rule = {};

export type Options = {
  activePage: number,
  itemsPerPage: number,
  sort: {
    field: string,
    direction: Direction,
  },
  queryRules: Array<Rule>,
};

export type Result = {
  items: Array<Object>,
  total: number,
};

export const encode = (target: Object): string => Buffer.from(JSON.stringify(target)).toString('base64');

export const DEFAULT_OPTIONS: Options = {
  activePage: 1,
  itemsPerPage: 20,
  sort: {
    field: 'id',
    direction: 'descending',
  },
  queryRules: [],
};

export const DEFAULT_RESULT: Result = {
  total: 0,
  items: [],
};
