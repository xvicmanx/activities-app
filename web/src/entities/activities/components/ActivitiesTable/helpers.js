// @flow

import moment from 'moment';

export const dateValueTableValueResolver = (item: Object): any => moment(item.date).format('DD-MM-YYYY HH:mm');

