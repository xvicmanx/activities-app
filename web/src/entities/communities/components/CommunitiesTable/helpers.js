// @flow

import React from 'react';
import { Link } from 'react-router-dom';

export const nameTableValueResolver = (item: Object): React$Element<any> => (
  <Link to={`/communities/${item.id}`}>{item.name}</Link>
);
