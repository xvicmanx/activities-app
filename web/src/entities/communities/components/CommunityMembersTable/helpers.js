// @flow

export const coordinatesTableValueResolver = (item: Object): string =>
  item.coordinates ? 'Si' : 'No';
