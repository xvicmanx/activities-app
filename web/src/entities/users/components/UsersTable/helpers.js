// @flow

export const throwErrorWhenNotSuccess = (fn: Function): ((item: any) => Promise<any>) => async (item) => {
  const result = await fn(item);

  if (!result.success) {
    throw new Error(result.message);
  }

  return result;
};
