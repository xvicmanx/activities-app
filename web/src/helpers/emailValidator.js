const emailValidator = (email) => {
  const mailformat = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/;

  return email.match(mailformat) ? true : false;
};

export default emailValidator;
