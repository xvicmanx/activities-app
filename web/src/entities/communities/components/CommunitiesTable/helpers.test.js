import ShallowRenderer from 'react-test-renderer/shallow';

import { nameTableValueResolver } from './helpers';

describe('CommunitiesTable helpers', () => {
  describe('nameTableValueResolver', () => {
    it('returns the expected values', () => {
      expect(nameTableValueResolver({ id: 1, name: 'John' })).toMatchSnapshot();
    });
  });
});
