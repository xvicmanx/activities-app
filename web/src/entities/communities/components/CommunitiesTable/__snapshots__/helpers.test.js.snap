// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunitiesTable helpers nameTableValueResolver returns the expected values 1`] = `
<ForwardRef
  to="/communities/1"
>
  John
</ForwardRef>
`;
